### ISSUE TITLE
Service based on GBM model can't be consumed

### ISSUE THREAD

**Issue by lixzhang** [Azure/Azure-MachineLearning-ClientLibrary-Python#22]
After I published a web service based on a GBM model, the service can't be consumed. Below is the code I used to set up the service, along with a screenshot of the error when calling the service. I did not have this problem when publishing a service based on a linear model. Any insights will be greatly appreciated.

    from azureml import Workspace
    from sklearn.ensemble import GradientBoostingRegressor
    ws = Workspace(

     workspace_id='b2bbeb56a1d04e1599d2510a06c59d87',
     authorization_token='57207d50feb8415a90406136e23c1a22',
     endpoint='https://studioapi.azureml.net'
    )
    experiment = ws.experiments['b2bbeb56a1d04e1599d2510a06c59d87.f-id.911630d13cbe4407b9fe408b5bb6ddef']
    ds = experiment.get_intermediate_dataset(
     node_id='a0a931cf-9fb3-4cb9-83db-f48211be560c-323',
     port_name='Results dataset',
     data_type_id='GenericCSV'
    )
    frame = ds.to_dataframe()

    mydata = frame

    # create X and y
    feature_cols = ['crim', 'zn', 'indus', 'chas', 'nox', 'rm', 'age', 'dis', 'rad', 'tax', 'ptratio', 'black', 'lstat']
    X = mydata[feature_cols]
    y = mydata.medv

    # fit model with the best set of parameter values
    params = {'n_estimators': 500, 'max_depth': 2, 'min_samples_split': 4,
              'learning_rate': 0.1, 'loss': 'ls', 'random_state': 0}
              
    gbm = GradientBoostingRegressor(**params)

    gbm.fit(X, y)

    # set up web service
    from azureml import services
    @services.publish('b2bbeb56a1d04e1599d2510a06c59d87', '57207d50feb8415a90406136e23c1a22')
    @services.types(crim=float, zn=float, indus=float, chas=float, nox=float, rm=float, 
                    age=float, dis=float, rad=float, tax=float, ptratio=float, black=float, lstat=float)
    @services.returns(float)
    def mygbm(crim, zn, indus, chas, nox, rm, age, dis, rad, tax, ptratio, black, lstat):
        # predict the label
        feature_vector = [crim, zn, indus, chas, nox, rm, age, dis, rad, tax, ptratio, black, lstat]
        return gbm.predict(feature_vector)

    # information about the web service
    print("url: " + mygbm.service.url + "\n")
    print("api_key: " + mygbm.service.api_key + "\n")
    print("help_url: " + mygbm.service.help_url + "\n")
    print("service id: " + mygbm.service.service_id + "\n")   

The screenshot:
![capture](https://cloud.githubusercontent.com/assets/9322661/11309055/a9fd6f42-8f8f-11e5-874e-a69e3778ad11.PNG)

**Comment by DinoV** [comment]
This looks to be due to incompatibilities between the version of scikit-learn we have on the backend AzureML services and the notebook service.  We'll be pushing out an updated version which has the correct version.  For the time being you can put this at the top of your notebook to install the same version:

```
!conda install -f scikit-learn=0.15.1 --yes
```

And run it before importing sklearn.

**Comment by DinoV** [comment]
Also I would encourage you to go and reset your authorization token (Settings -> Authorization Tokens -> Regenerate) in AzureML because they have been leaked by including them in the bug report.  I've removed them from the initial issue.

**Comment by lixzhang** [comment]
Yes, this solved the problem for me. Thank you.
Are there plans to make sure that the backend AzureML services and the notebook service be compatible in future? Making them compatible could help avoid many problems similar to this one.

**Comment by gilbertw** [comment]
Yes, we are aggressively working to standardize the two environments.  Thanks again for taking the time to file this issue
